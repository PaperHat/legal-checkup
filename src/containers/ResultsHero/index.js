import React from 'react'
import { connect } from 'react-redux'

import { getYesAnswerCount } from '@state/selectors'
import StyledResultsHero from '@components/styled/ResultsHero'

function ResultsHero ({ yesCount }) {
  return (
    <StyledResultsHero>
      <p>
        {yesCount > 0
          ? `Based on your answers, a lawyer may be able to help you with...`
          : `Based on your answers, it looks like you're in good legal health! 
        But if you have a problem, or something worrying you that wasn't covered by the questions 
        in this check-up, do come talk to us at Community Law`}

        {console.log('YesCount:', yesCount)}
      </p>
      {yesCount > 0 ? <a href='#'>expand</a> : null}
    </StyledResultsHero>
  )
}

function mapStateToProps (state) {
  return {
    yesCount: getYesAnswerCount(state)
  }
}

export default connect(mapStateToProps)(ResultsHero)
