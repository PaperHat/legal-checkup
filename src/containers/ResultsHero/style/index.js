import styled from 'styled-components'

const ResultsHero = styled.div``

export default ResultsHero
