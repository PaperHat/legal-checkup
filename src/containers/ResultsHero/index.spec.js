import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import { state } from '@state/fixtures'

const mockStore = configureStore()

const store = mockStore(state)

import ResultsHero from '.'

describe(`containers:ResultsHero`, () => {
  describe(`ResultsHero`, () => {
    it(`should display expand controller when yesCount > 0`, () => {
      expect(toJson(shallow(<ResultsHero store={store} />))).toMatchSnapshot()
    })

    it(`should display ResultsHero`, () => {
      expect(
        toJson(shallow(<ResultsHero store={store} />).dive())
      ).toMatchSnapshot()
    })

    it(`should map yesCount to ResultsHero props`, () => {
      expect(shallow(<ResultsHero store={store} />).props().yesCount).toBe(1)
    })
  })

  describe(`MapStateToProps`, () => {
    it(`should map yesCount to ResultsHero props`, () => {
      expect(shallow(<ResultsHero store={store} />).props().yesCount).toBe(1)
    })
  })
})
