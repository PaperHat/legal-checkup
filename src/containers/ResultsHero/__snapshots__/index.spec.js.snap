// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers:ResultsHero ResultsHero should display ResultsHero 1`] = `
<styled.div>
  <p>
    Based on your answers, a lawyer may be able to help you with...
  </p>
  <a
    href="#"
  >
    expand
  </a>
</styled.div>
`;

exports[`containers:ResultsHero ResultsHero should display expand controller when yesCount > 0 1`] = `
<ResultsHero
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  yesCount={1}
/>
`;
